modules/src/core/ics04_channel/handler.rs

```diff
@@ -39,38 +40,235 @@ pub struct ChannelResult {
     pub port_id: PortId,
     pub channel_id: ChannelId,
     pub channel_id_state: ChannelIdState,
-    pub channel_cap: Capability,
+    pub channel_cap: ChannelCapability,
     pub channel_end: ChannelEnd,
 }
 
+pub fn channel_validate<Ctx>(ctx: &Ctx, msg: &ChannelMsg) -> Result<ModuleId, Error>
+where
+    Ctx: Ics26Context,
+{
+    let module_id = match msg {
```

Comment:
We can implement a method for `ChannelMsg` that returns `ModuleId`.

Reply:
https://github.com/informalsystems/ibc-rs/pull/1838/commits/2cc6b71793c743365098a616dbead9e95bfd56fe